doc/api/cli.md

```diff
@@ -892,6 +892,94 @@ generated as part of the test runner output. If no tests are run, a coverage
 report is not generated. See the documentation on
 [collecting code coverage from tests][] for more details.
 
+### `--check-coverage`
+
+<!-- YAML
+added:
+  - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+The `--check-coverage` CLI flag, used in conjunction with
+the `--experimental-test-coverage` command, enforces that
+the test coverage thresholds specified by check
+flags are respected:
+
+* [`--lines=threshold`][]: Sets the threshold for line coverage.
+* [`--branches=threshold`][]: Sets the threshold for branch coverage.
+* [`--functions=threshold`][]: Sets the threshold for function coverage.
+
+To enforce global 100% coverage set the value of the each flag to 100:
+
+```bash
+node --test --experimental-test-coverage --check-coverage --lines=100 --branches=100 --functions=100
+```
+
+If one of the specified coverage check falls below the threshold,
+the test will exit with non zero code.
+
+### `--lines=threshold`
```

Comment:
> I still think that, if folks keep adding more and more flags for the test runner, they should at least be scoped, i.e., the name should be somewhat specific (e.g., `--test-coverage-line-threshold`) or the option should only be parsed in the context of the test runner (similar to python's `-m`, for example). Otherwise, we are just bloating the CLI with ambiguously named options.

I do agree that flags need to be scoped but also not extremely long. Can you elaborate on how the option should be parsed in test runner context?

Reply:
One idea may be to let `--check-coverage` take a string argument with the additional options, for example:

```sh
--check-coverage=lines=80,branches=100

# or
--check-coverage=lines:80,branches:100

# or support passing multiple times, one aspect each
--check-coverage lines=80 --check-coverage branches=100
```

I know this increases parsing complexity and may not be aligned with other CLI flags in Node.js (though I haven't checked them all). There is, however, precedence in other CLIs such as the [Docker CLI](https://docs.docker.com/engine/reference/commandline/run/#env):

```sh
$ docker run --env VAR1=value1 --env VAR2=value2 ubuntu
```